### ISSUE TITLE
Item thumbnails do not initially display after running filter-media

### ISSUE THREAD

**Issue by carlj11** [DSpace/dspace-angular#797]
Not sure if this is a bug or if I'm missing some configuration parameter or page setting (I'm new to angular) that determines thumbnail display for simple and full item views, or if this is functionality that hasn't been fully implemented yet.  After running filter-media for items show an empty icon with 'No Thumbnail' and I can only get thumbnails to display after  I've gone from simple item page view to full item page and then back again to simple item page view.   Thumbnails are for jpg's and .pdf's and happens with each browser session. 

I'm on the dspace-angular and rest-api snapshots from 7/14.  

Carl

**Comment by tdonohue** [comment]
@carlj11 : A quick question about what you are seeing.   

It *sounds like* you **are** able to get Thumbnails to appear, but only after leaving the page and going back again?  Is that correct?

If so, I highly suspect this is a **caching issue** in the Angular UI (and not a configuration issue or anything).  It sounds like the UI is caching the old "No Thumbnail" empty icon, and only _reloads_ that cache after you leave the page and go back again.

If I've misunderstood, please let me know.  I'm trying to determine exactly what you are seeing though, so that I know how to schedule this bug ticket.

**Comment by carlj11** [comment]
Hi Tim,

Yes, thumbnails appear if I leave the first page and come back again.  In this case the simple page view, go to full item view, and then back to simple view. At that point the thumbnails appear in either view.   If I leave and come back to an item I have to repeat the process again. 

BTW, this is a locally installed instance on a Mac Pro w/Catalina, 10.15.5.  I’m not running Docker at the moment. 

Thanks.

Carl

**Comment by carlj11** [comment]
I should also add this is only when coming from list view of items.   It doesn’t seem to happen if I search and am given the option to switch to grid (gridElement) view to see the results.

FWIW, would be nice to have the ability to toggle between list and grid views not only be limited to search results. One of our instances has mostly visual content and any kind of browse view would benefit from the option.  

Carl

**Comment by carlj11** [comment]
Revise my last statement.  Page refresh is still needed to display thumbnails even in grid view.  My mistake, I had visited an item previously.

**Comment by tdonohue** [comment]
@carlj11 : This sounds suspiciously like a caching issue in the UI, especially if it works OK after visiting an item previously.

One final question, what browser are you using?  I'll see if we can find someone else who can reproduce the bug, and if so, hopefully we can figure out why the thumbnails are not loading immediately.

**Comment by artlowel** [comment]
This is indeed due to the cache, and doesn't seem like a bug to me. The item will be cached by default for 15 minutes after which it will be reloaded (once #739 is complete anyway) and the thumbnail will be shown. 

There's no way the UI can know you've run filter-media in the mean time, not unless we add ETag support or something similar. So for now, if an admin knows they've run filter-media and want to see the result immediately they can always do a hard refresh. Other users will see it at most 15 minutes later.

**Comment by tdonohue** [comment]
@artlowel : Thanks for the response here.  Based on your feedback, I'm going to close this as "won't fix".

I agree that it's OK if thumbnails are cached temporarily (as it's for performance reasons), and that caching thumbnails for 15mins or until a hard refresh (Ctrl + Refresh in user's browser) is done is perfectly OK.
